### PR TITLE
Fix even more entity metadata flags

### PR DESCRIPTION
**Pull request by rtm516**
Fixes the following metadata flags for entities:

- Fox
  - INTERESTED
- Rabbit
  - VARIANT - Colors of the Rabbit
- Horse
  - DISPLAY_ITEM - Used for opening the mouth
  - EATING_ITEM (event) - Displays the eating particles
- Bat
  - RESTING - If the Bat is hanging upside down
- Snow Golem
  - SHEARED - Visibility of the pumpkin
- Mooshroom
  - VARIANT - Color of the Mooshroom (red or brown)
- Turtle
  - IS_PREGNANT - If they are carrying an egg
  - LAYING_EGG
- Vex
  - TARGET_EID - Shows the attack animation

### CODE REVIEW

File: connector/src/main/java/org/geysermc/connector/entity/living/animal/horse/AbstractHorseEntity.java [hunk 1]
```diff
@@ -47,6 +51,18 @@ public void updateBedrockMetadata(EntityMetadata entityMetadata, GeyserSession s
             metadata.getFlags().setFlag(EntityFlag.SADDLED, (xd & 0x04) == 0x04);
             metadata.getFlags().setFlag(EntityFlag.EATING, (xd & 0x10) == 0x10);
             metadata.getFlags().setFlag(EntityFlag.STANDING, (xd & 0x20) == 0x20);
+
+            // Bedrock uses DISPLAY_ITEM to work out if the mouth is open
+            metadata.put(EntityData.DISPLAY_ITEM, (xd & 0x40) == 0x40 ? 128 : 0);
```
Comment (DoctorMacc) [inline connector/src/main/java/org/geysermc/connector/entity/living/animal/horse/AbstractHorseEntity.java:56]: Is 128 supposed to be the wheat ID?

Comment (rtm516) [reply connector/src/main/java/org/geysermc/connector/entity/living/animal/horse/AbstractHorseEntity.java:56]: Seems to be a magic number, maybe its time or something. Trying with 2 different foods for the horse gives the same number.

Comment (DoctorMacc) [reply connector/src/main/java/org/geysermc/connector/entity/living/animal/horse/AbstractHorseEntity.java:56]: If we can't work out what it is, then put a note saying that.

File: connector/src/main/java/org/geysermc/connector/entity/living/animal/horse/AbstractHorseEntity.java [hunk 2]
```diff
@@ -47,6 +51,28 @@ public void updateBedrockMetadata(EntityMetadata entityMetadata, GeyserSession s
...
+            int horseFlags = 0x0;
+            horseFlags = (xd & 0x40) == 0x40 ? horseFlags | 0x80 : horseFlags;
+
+            // Only set eating when we don't have mouth open so a player interaction doesn't trigger the eating animation
+            horseFlags = (xd & 0x10) == 0x10 && (xd & 0x40) != 0x40 ? horseFlags | 0x20 : horseFlags;
+
+            // Set the flags into the display item
+            metadata.put(EntityData.DISPLAY_ITEM, horseFlags);
+
+            // Send the eating particles
```
Comment (DoctorMacc) [inline connector/src/main/java/org/geysermc/connector/entity/living/animal/horse/AbstractHorseEntity.java:68]: Add a note here about Bedrock/Java non-parity?

**Review by DoctorMacc — APPROVED** [review]
Just one minor comment, so I'm pre-approving.

**Pull request merged**